radicle-tracker/src/metadata.rs

```diff
@@ -0,0 +1,240 @@
+use std::{
+    collections::{HashMap, HashSet},
+    convert::Infallible,
+    hash::Hash,
+    str::FromStr,
+};
+
+/// The metadata that is related to an issue.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Metadata<User: Eq + Hash> {
+    labels: HashSet<Label>,
```

Comment:
Are we supposed to use labels to mark an issue as open or closed?

Reply:
_oof_ no forgot about closing an issue :sweat_smile: I'll add that :)

Reply:
This is not representing the stored data, right? I think we still haven't made up our minds how we want to deal with issue lifecycle.

If we think of an issue with attached code (a "PR" or "revision"), merging the code to some mainline branch would consequently entail merging the conversation as well (to some _other_ mainline branch? idk). But then, what if it gets rejected? Does that disappear from the project history? What does it mean to "reopen" an issue?

Anyways, since we're concerned with the _surface_ API here (not how it gets stored), maybe a good representation would be to wrap the issue in a newtype `ClosedIssue(Issue)` so as to restrict what the caller _can do_ with that issue. Obviously I haven't thought this through. Probably more complicated, like a `Lifecycle` FSM or something.

Reply:
Nah, this is just the materialized view of the data. The stored data will come to a head when we talk about the op-based semantics and possibly the Merkle-DAG stuff.

> Anyways, since we're concerned with the surface API here (not how it gets stored), maybe a good representation would be to wrap the issue in a newtype ClosedIssue(Issue) so as to restrict what the caller can do with that issue. Obviously I haven't thought this through. Probably more complicated, like a Lifecycle FSM or something.

This is pretty much what I was thinking too :)